### DISCUSSION TITLE
Question: I want to know details of kwok

### DISCUSSION THREAD

**Discussion by sivchari** [kubernetes-sigs/kwok#86] [Q&A] [answered]
I'm interesting at the kwok and I want to contribute more. 
So, I want to know more about the motivation and goal of kwok.
Thanks.

**Comment by wzshiming** [comment]
Thank you for your interest in Kwok

First, we will migrate the content of the two Repo's to the Kwok.
- https://github.com/wzshiming/fake-k8s
- https://github.com/wzshiming/fake-kubelet

Then there are some issues that will do
- #2
- #4
- #8
- #10
- #16

**Comment by sivchari** [comment]
Thanks,  I see.
I want to develop kwok, too.

**Reply by wzshiming** [reply]
https://github.com/kubernetes-sigs/kwok/issues/107

I created an issue to clean up the code, and if you are interested, you can continue to participate from there to help improve the quality of the code together.

**Reply by sivchari** [reply]
Thanks, but #107 will be dealt by good contributors. So, I would contribute other issues 👍

**Comment by qingwave** [comment]
Hi @wzshiming, 

Is there some primary use case for Kwok, for test or a toy kubernetes? I don't find detail documents regarding this.

**Reply by wzshiming** [reply]
KWOK is mainly used for control-plane performance testing and data simulation and as a replacement for Kind in tests that do not require actual running containers.

**Comment by wzshiming** [comment, marked as answer]
Relevant details will be added to https://kwok.sigs.k8s.io.
